### PR TITLE
Add tests for the intrinsic size of a video element

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/19189)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/283623446)

**Comment by nox** [comment]
I changed my tests a bit, note that two of them don't pass in Safari and Chrome, because it's too eager to update the dimensions before layout was done again.

**Comment by zcorpan** [comment]
Can you also please make a PR for presto-testo to remove the converted tests, and link to this PR?

**Comment by nox** [comment]
@zcorpan Done in https://github.com/operasoftware/presto-testo/pull/54.

**Comment by zcorpan** [comment]
Thanks!

I would do `async_test(function(t) {` and remove the nested `async_test`, but I will not object to merging as-is. So LGTM with or without this suggested change.

**Review by zcorpan — APPROVED** [review]

**Comment by gsnedders** [comment]
@nox are you planning on doing that? if not we should just land it

**Comment by nox** [comment]
I can do that, but I forgot about it.

**Comment by nox** [comment]
Done.

File: html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      assert_equals(s.height, "150px");
+    }));
+  });
+}, "default object size after src is removed");
+
+async_test(function(t) {
+  var v = document.getElementById("v5");
+  var s = getComputedStyle(v);
+  v.onerror = t.unreached_func();
+  onload = t.step_func(function() {
```
Comment (nox) [inline html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm]: I don't remember why that is just `onload`.

Comment (zcorpan) [reply html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm]: Same as `window.onload`, it's waiting for the poster image to load, I believe.

File: html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    v.load();
+    // Dimensions should be updated only on next layout.
+    assert_equals(s.width, '320px');
+    assert_equals(s.height, '240px');
+    requestAnimationFrame(t.step_func_done(function() {
+      assert_equals(s.width, "300px");
+      assert_equals(s.height, "150px");
+    }));
+  });
+}, "default object size after src is removed");
```
Comment (foolip) [inline html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm]: This and the last tests fails for me in Chrome, Firefox and Safari. Can you add descriptions to the assertions to make it easier to understand which assertion failed? I suspect this isn't right per spec, but it's quite hard to tell.

Comment (nox) [reply html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm]: It's the same thing as with all other kinds of elements that have intrinsic dimensions. This pull request is one year old and I've moved on to work on other stuff (WebGL), this stuff isn't fresh in my mind anymore. I'm not sure what you mean by "the last tests" btw.

Comment (nox) [reply html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm]: See https://github.com/operasoftware/presto-testo/pull/54/files#diff-965ac4475894b45c273738db54ddd8a2

Comment (zcorpan) [reply html/semantics/embedded-content/the-video-element/intrinsic_sizes.htm]: @foolip can we just merge this and follow up with a new issue/PR if it needs more changes?

**Pull request merged**